File: recursive_addition/perl_recursive_addition/recursive_addition.pl
#!/usr/bin/perl -w

use strict;
use warnings;

sub rec_add {
  my $x = shift;
  my $y = shift;

  if( $y == 0 ) {
    return $x;
  } else {
    return 1 + rec_add( $x, $y - 1 );
  }
}


sub main {
  if( $#ARGV+1 != 0 ) {
    print STDERR "Usage: $0 <noargs>\n";
    exit(1);
  }
  #print $ARGV[0], "\n";
  print "rec_add(6,8)=" . &rec_add(6,8) . "\n";
  exit(0);
}

&main();

File: recursive_addition/ocaml_recursive_addition/recursive_addition.ml
let rec_add x  y =
    if y = 0 then
      x
    else
      1 + rec_add(x,(y - 1));;

let main () =
  if  Array.length Sys.argv <> 1 then
    begin
      Printf.printf "%s <noargs>\n" Sys.argv.(0);
      exit 1
    end
  else
    begin
      print_endline "begining of program"
    end

let _ = main ()
